addons/hr/models/res_users.py

```diff
@@ -1,21 +1,189 @@
+    country_of_birth = fields.Many2one(related='employee_ids.country_of_birth', readonly=False, related_sudo=False)
+    marital = fields.Selection(related='employee_ids.marital', readonly=False, related_sudo=False)
+    spouse_complete_name = fields.Char(related='employee_ids.spouse_complete_name', readonly=False, related_sudo=False)
+    spouse_birthdate = fields.Date(related='employee_ids.spouse_birthdate', readonly=False, related_sudo=False)
+    children = fields.Integer(related='employee_ids.children', readonly=False, related_sudo=False)
+    emergency_contact = fields.Char(related='employee_ids.emergency_contact', readonly=False, related_sudo=False)
+    emergency_phone = fields.Char(related='employee_ids.emergency_phone', readonly=False, related_sudo=False)
+    visa_no = fields.Char(related='employee_ids.visa_no', readonly=False, related_sudo=False)
+    permit_no = fields.Char(related='employee_ids.permit_no', readonly=False, related_sudo=False)
+    visa_expire = fields.Date(related='employee_ids.visa_expire', readonly=False, related_sudo=False)
+    google_drive_link = fields.Char(related='employee_ids.google_drive_link', readonly=False, related_sudo=False)
+    additional_note = fields.Text(related='employee_ids.additional_note', readonly=False, related_sudo=False)
+    barcode = fields.Char(related='employee_ids.barcode', readonly=False, related_sudo=False)
+    pin = fields.Char(related='employee_ids.pin', readonly=False, related_sudo=False)
+
+    readable_fields = [
+        'employee_ids',
+        'active',
+        'display_name',
+        'job_title',
+        'work_phone',
+        'phone',
+        'mobile_phone',
+        'work_email',
+        'category_ids',
+        'department_id',
+        'address_id',
+        'work_location',
+        'parent_id',
+        'coach_id',
+        'notes',
+        'address_home_id',
+        'is_address_home_a_company',
+        'km_home_work',
+        'bank_account_id',
+        'country_id',
+        'identification_id',
+        'passport_id',
+        'gender',
+        'birthday',
+        'place_of_birth',
+        'country_of_birth',
+        'marital',
+        'spouse_complete_name',
+        'spouse_birthdate',
+        'children',
+        'emergency_contact',
+        'emergency_phone',
+        'visa_no',
+        'permit_no',
+        'visa_expire',
+        'google_drive_link',
+        'additional_note',
+        'child_ids',
+        'parent_id',
+        'pin',
+        'barcode',
+    ]
+
+    def __init__(self, pool, cr):
+        """ Override of __init__ to add access rights.
+            Access rights are disabled by default, but allowed
+            on some specific fields defined in self.SELF_{READ/WRITE}ABLE_FIELDS.
+        """
+        init_res = super(User, self).__init__(pool, cr)
+        # duplicate list to avoid modifying the original reference
+        type(self).SELF_WRITEABLE_FIELDS = list(self.SELF_WRITEABLE_FIELDS)
```

Comment:
How would you do it in the body of the class ?
What do you thing of the override below ?
```python
def __new__(cls, *args, **kwargs):
        """ Override of __new__ to add access rights.
            Access rights are disabled by default, but allowed
            on some specific fields defined in self.SELF_{READ/WRITE}ABLE_FIELDS.
        """
        cls_instance = super(User, cls).__new__(cls, *args, **kwargs)
        # duplicate list to avoid modifying the original reference
        cls_instance.SELF_READABLE_FIELDS = cls_instance.hr_expense_fields + cls_instance.SELF_READABLE_FIELDS
        return cls_instance
```
`*args` and `**kwargs` are not necessary, I'm not sure it's a good idea to have them or not